src/Language/Syntax.hs

```diff
@@ -3,75 +3,52 @@
 -- this slightly deviates from the specified syntax in spots.
 
 module Language.Syntax where
+
 import Language.Types
 import Data.List
 import GHC.Generics
 import Data.Aeson
 import Data.Array
 import qualified Data.Set as S
+
 type Name = String
 
 -- | Game datatype
 data Game a = Game
   {
-    name  :: Name -- ^ The name of the game
-  , board :: BoardDef -- ^ Size and type of the board
-  , input :: InputDef -- ^ Type of input
-  , defns ::[ValDef a] -- ^ List of value definitions
+    name  :: Name       -- ^ The name of the game
+  , board :: BoardDef   -- ^ Size and type of the board
+  , input :: InputDef   -- ^ Type of input
+  , defns :: [ValDef a] -- ^ List of value definitions
   }
   deriving (Generic)
 
-instance Show (Game a) where
```

Comment:
I moved all the show instances to the bottom of the file. I did this because I think it is best to have the more meaningful bits of the syntax grouped together. I often find myself referencing this file and it's not ergonomic to jump around it.

Reply:
👍